tutor/src/components/course-listing/listings.cjsx

```diff
@@ -0,0 +1,155 @@
+_ = require 'lodash'
+React = require 'react'
+BS = require 'react-bootstrap'
+{Link} = require 'react-router'
+classnames = require 'classnames'
+
+Router = require '../../helpers/router'
+
+{CourseStore} = require '../../flux/course'
+{CurrentUserStore} = require '../../flux/current-user'
+
+CourseData = require '../course-data-mixin'
+IconAdd = require  '../icons/add'
+
+{Course, CourseTeacher} = require './course'
+
+getReactBaseName = (context) -> _.kebabCase(context.constructor.displayName)
+
+wrapCourseItem = (Item, course = {}) ->
+  {id} = course
+  courseName = id or 'new'
+
+  if id
+    courseDataProps = CourseData.getCourseDataProps(id)
+    courseSubject = CourseStore.getSubject(id)
+    courseIsTeacher = CourseStore.isTeacher(id)
+
+  <BS.Col key="course-listing-item-wrapper-#{courseName}" md={3} sm={6} xs={12}>
+    <Item
+      course={course}
+      courseSubject={courseSubject}
+      courseIsTeacher={courseIsTeacher}
+      courseDataProps={courseDataProps}/>
+  </BS.Col>
+
+DEFAULT_COURSE_ITEMS =
+  teacher: CourseTeacher
+  student: Course
+
+CourseListingBase = React.createClass
+  displayName: 'CourseListingBase'
+
+  getItems: ->
+    _.merge({}, DEFAULT_COURSE_ITEMS, @props.Items)
+
```

Comment:
Why's the prop `Item` capitalized?  Also, the classes in the file are missing propTypes 😁